modules/convert.py

```diff
@@ -339,17 +495,20 @@ def update_cache(cache_ids, id_type, imdb_in, guid_type):
                     logger.info(f" Cache  |  {'^' if expired else '+'}  | {ids} | {item.title}")
                     self.config.Cache.update_guid_map(item.guid, cache_ids, imdb_in, expired, guid_type)
 
-            if (tmdb_id or imdb_id) and library.is_movie:
-                update_cache(tmdb_id, "TMDb", imdb_id, "movie")
-                return "movie", tmdb_id, imdb_id
+            if (tmdb_movie_id or imdb_id) and library.is_movie:
+                update_cache(tmdb_movie_id, "TMDb", imdb_id, "movie")
+                return "movie", tmdb_movie_id, "TMDb", imdb_id
             elif (tvdb_id or imdb_id) and library.is_show:
                 update_cache(tvdb_id, "TVDb", imdb_id, "show")
-                return "show", tvdb_id, imdb_id
-            elif anidb_id and (tmdb_id or imdb_id) and library.is_show:
-                update_cache(tmdb_id, "TMDb", imdb_id, "show_movie")
-                return "movie", tmdb_id, imdb_id
+                return "show", tvdb_id, "TVDb", imdb_id
+            elif anidb_id and tmdb_movie_id and library.is_show:
+                update_cache(tmdb_movie_id, "TMDb", imdb_id, "show_movie")
+                return "movie", tmdb_movie_id, "TMDb", imdb_id
+            elif tmdb_show_id and library.is_show:
+                update_cache(tmdb_show_id, "TMDb", imdb_id, "show_tmdb")
```

Comment:
`show_tmdb` isn't a great name, but it gets the point across.

Note: This will only happen if the show doesn't match in both IMDb and TVDb, but has a TMDb entry which is uncommon.